From 6aba6d330f4275fa101ad0d07852d6d8e701d8a6 Mon Sep 17 00:00:00 2001
From: "Liu,Dalei" <daleix.liu@intel.com>
Date: Sun, 21 Dec 2014 22:34:09 +0800
Subject: [PATCH 184/196] atomisp2: Fix a memory leak problem

It will lead memory leak, if vmap failed.

Change-Id: Ia99017e4feb25aa8b058c1971a9bc7a3442bccbf
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
Tracked-on: https://jira01.devtools.intel.com/browse/IMINAN-14504
Reviewed-on: https://icggerrit.ir.intel.com/33744
Reviewed-by: Shuguang Gong <shuguang.gong@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c        | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c
index 60c4de9..19edf29 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c
@@ -1106,6 +1106,7 @@ void *hmm_bo_vmap(struct hmm_buffer_object *bo, bool cached)
 
 	bo->vmap_addr = vmap(pages, bo->pgnr, VM_MAP, cached ? PAGE_KERNEL : PAGE_KERNEL_NOCACHE);
 	if (unlikely(!bo->vmap_addr)) {
+		atomisp_kernel_free(pages);
 		mutex_unlock(&bo->mutex);
 		dev_err(atomisp_dev, "vmap failed...\n");
 		return NULL;
-- 
1.9.1

